smaug/python/graph.py

```diff
@@ -102,32 +102,29 @@ def add_node(
           dims=d, name="%s/output%d" % (node.name, i),
           data_layout=output_tensor_layout, data_type=output_tensor_dtype,
           data_format=output_tensor_dformat, source=(node, i),
-          alignment=self.alignment)
-      output_tensor_proto = node.output_tensors.add()
-      output_tensor.to_tensor_proto(output_tensor_proto, self.tensor_data_array)
+          alignment=self._alignment)
+      node.add_output(output_tensor)
       output_tensors.append(output_tensor)
 
     return output_tensors
 
-  def get_node(self, node_name, recursive=False):
-    """Return a node in the graph proto by its name.
+  def get_node(self, node_name):
+    """Return a node in the graph by its name.
 
     Args:
       node_name: Node name.
-      recursive: If true, recursively search the node in the parent graphs.
 
     Returns:
-      A NodeProto if we find the node.
+      A `Node` if we find the node or None is returned.
     """
-    for i in range(len(self.graph.nodes)):
-      if self.graph.nodes[i].name == node_name:
-        return self.graph.nodes[i]
-    if recursive and self._parent_graph is not None:
-      return self._parent_graph.get_node(node_name, True)
+    for node in self._nodes:
```

Comment:
Why did we remove the recursive search to the parents?

Reply:
I removed it because there's no use case for now. But it's potential useful. Added it back.